scapy/contrib/cansocket_native.py

```diff
@@ -13,30 +13,45 @@
 import struct
 import socket
 import time
+
 from scapy.config import conf
 from scapy.supersocket import SuperSocket
 from scapy.error import Scapy_Exception, warning
+from scapy.packet import Packet
 from scapy.layers.can import CAN, CAN_MTU
-from scapy.packet import Padding
 from scapy.arch.linux import get_last_packet_timestamp
+from scapy.compat import List, Dict, Type, Any, Optional, Tuple, raw
 
 conf.contribs['NativeCANSocket'] = {'channel': "can0"}
 
 
 class NativeCANSocket(SuperSocket):
+    """Initializes a Linux PF_CAN socket object.
+
+    Example:
+        >>> socket = NativeCANSocket(channel="vcan0", can_filters=[{'can_id': 0x200, 'can_mask': 0x7FF}])  # noqa: E501
+
+    :param channel: Network interface name
+    :param receive_own_messages: Messages, sent by this socket are will
+                                 also be received.
+    :param can_filters: A list of can filter dictionaries.
+    :param basecls: Packet type in which received data gets interpreted.
+    :param kwargs: Various keyword arguments for compatibility with
+                   PythonCANSockets
+    """
     desc = "read/write packets at a given CAN interface using PF_CAN sockets"
 
     def __init__(self, channel=None, receive_own_messages=False,
-                 can_filters=None, remove_padding=True, basecls=CAN, **kwargs):
```

Comment:
I advise you to use the
```
def __init__(self,
                   channel=None,  # type: something
                   receive_own_messages=False,  # type: something
                   ...
                   ):
    # type: (...) -> None
```
format, as done very often in the core to avoid noqas (and generally make longer lines clearer)

Reply:
I will apply that, and also use that in future PRs